Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -9,54 +9,57 @@ public GildedRose(Item[] items) {
 
     public void updateQuality() {
         for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
+            StoredItem currentItem = new StoredItem(items[i]);
+            if (itemCanLoseQuality(currentItem)) {
+                currentItem.decreaseQuality();
             } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
+                if (isLowQuality(currentItem)) {
+                    currentItem.increaseQuality();
 
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
+                    if (currentItem.isNamed("Backstage passes to a TAFKAL80ETC concert")) {
+                        if (currentItem.sellIn < 11) {
+                            if (isLowQuality(currentItem)) {
+                                currentItem.increaseQuality();
                             }
                         }
 
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
+                        if (currentItem.sellIn < 6) {
+                            if (isLowQuality(currentItem)) {
+                                currentItem.increaseQuality();
                             }
                         }
                     }
                 }
             }
 
-            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                items[i].sellIn = items[i].sellIn - 1;
+            if (currentItem.isNotNamed("Sulfuras, Hand of Ragnaros")) {
```

Comment:
For sulfuras there's nothing to do, maybe an early return / continue would be more understandable.